modules/battenberg/1.1/battenberg.smk

```diff
@@ -50,11 +50,48 @@ _battenberg_CFG = CFG
 
 # Define rules to be run locally when using a compute cluster
 localrules:
+    _battenberg_get_reference
     _battenberg_all
 
-
 ##### RULES #####
 
+# Downloads the reference files into the module results directory (under '00-inputs/') from https://www.bcgsc.ca/downloads/morinlab/reference/ . 
+rule _battenberg_get_reference:
+    output:
+        battenberg_impute =  directory(CFG["dirs"]["inputs"] + "reference/{genome_build}/battenberg_impute_v3"),
+        impute_info = CFG["dirs"]["inputs"] + "reference/{genome_build}/impute_info.txt",
+        probloci =  CFG["dirs"]["inputs"] + "reference/{genome_build}/probloci.txt.gz",
+        battenberg_wgs_replic_correction = directory(CFG["dirs"]["inputs"] + "reference/{genome_build}/battenberg_wgs_replic_correction_1000g_v3"),
+        battenberg_gc_correction = directory(CFG["dirs"]["inputs"] + "reference/{genome_build}/battenberg_wgs_gc_correction_1000g_v3"),  
+        genomesloci = directory(CFG["dirs"]["inputs"] + "reference/{genome_build}/battenberg_1000genomesloci2012_v3")
+    params:
+        url = "https://www.bcgsc.ca/downloads/morinlab/reference",
+        folder = CFG["dirs"]["inputs"] + "reference/{genome_build}",
+        build = lambda w: "hg38" if "38" in str({w.genome_build}) else "grch37",
+        PATH = CFG['inputs']['src_dir']
+    shell:
+        op.as_one_line("""
+        wget -qO-  {params.url}/battenberg_impute_{params.build}.tar.gz  |
+        tar -xvz > {output.battenberg_impute} -C {params.folder}
+        &&
+        wget -qO- {params.url}/battenberg_{params.build}_gc_correction.tar.gz  |
+        tar -xvz > {output.battenberg_gc_correction} -C {params.folder}
+        &&
+        wget -qO- {params.url}/battenberg_1000genomesloci_{params.build}.tar.gz | 
+        tar -xvz > {output.genomesloci} -C {params.folder}
+        &&
+        wget -O {output.impute_info} 'https://ora.ox.ac.uk/objects/uuid:2c1fec09-a504-49ab-9ce9-3f17bac531bc/download_file?file_format=plain&safe_filename=impute_info.txt&type_of_work=Dataset'
+        &&
+        python {params.PATH}/reference_correction.py {params.build}
```

Comment:
Because this rule uses python it should use a conda environment with the appropriate python version.